api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -286,7 +286,7 @@ func (m *AzureManagedMachinePool) validateMaxPods() error {
 
 func (m *AzureManagedMachinePool) validateOSType() error {
 	if m.Spec.Mode == string(NodePoolModeSystem) {
-		if m.Spec.OSType != nil && *m.Spec.OSType != azure.LinuxOS {
+		if m.Spec.OSType != nil && *m.Spec.OSType != "Linux" {
```

Comment:
Maybe "Linux" and "Windows" should be local constants instead of hard-coded?

Reply:
done